### ISSUE TITLE
Android TV loses connection after wake from sleep - Internal Server Error

### ISSUE THREAD

**Issue by Yezariael75** [unfoldedcircle/feature-and-bug-tracker#34]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I added Nvidia Shield over Android TV integration. Works fine but if remote goes to sleep and wakes up connection to network is lost. Only solution is to reboot the device.

### How to Reproduce

1. Add Android TV integration with Nvidia Shield
2. Start activity for Nvidia Shield (Plex)
3. Wait until remote goes to sleep
4. press a button - connection lost and does no get established again until reboot of device -> Message "Internal Server Error"

I assume the android tv integration crashes in the background?

### Expected behavior

Remote wakes up and establishes connection, can be used as before sleep.

### System version

1.1.1

### What part of the system affected by the problem?

Integration

**Comment by vistalba** [comment]
I run into the same issue. Also nVidia Shield. Quite annoying.

**Comment by splattner** [comment]
not sure if I see the same issue here. I was able to add my Android TV and initially it worked and I could use it. But then suddenly, it stopped.
Not seeing anything in the log:
```
2023-09-11 12:58:07.792542 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 109, "code": 200, "msg": "result", "msg_data": {}}
2023-09-11 12:58:07.787115 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":109,"msg":"entity_command","msg_data":{"cmd_id":"cursor_up","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
2023-09-11 12:58:07.332477 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 108, "code": 200, "msg": "result", "msg_data": {}}
2023-09-11 12:58:07.326893 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":108,"msg":"entity_command","msg_data":{"cmd_id":"cursor_down","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
2023-09-11 12:58:06.647171 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 107, "code": 200, "msg": "result", "msg_data": {}}
2023-09-11 12:58:06.641784 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":107,"msg":"entity_command","msg_data":{"cmd_id":"cursor_down","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
2023-09-11 12:58:06.176987 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 106, "code": 200, "msg": "result", "msg_data": {}}
2023-09-11 12:58:06.171262 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":106,"msg":"entity_command","msg_data":{"cmd_id":"cursor_left","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
2023-09-11 12:58:05.508621 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 105, "code": 200, "msg": "result", "msg_data": {}}
```

Rebooting the remote, also helped.

**Comment by martonborzak** [comment]
> not sure if I see the same issue here. I was able to add my Android TV and initially it worked and I could use it. But then suddenly, it stopped. Not seeing anything in the log:
> 
> ```
> 2023-09-11 12:58:07.792542 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 109, "code": 200, "msg": "result", "msg_data": {}}
> 2023-09-11 12:58:07.787115 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":109,"msg":"entity_command","msg_data":{"cmd_id":"cursor_up","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
> 2023-09-11 12:58:07.332477 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 108, "code": 200, "msg": "result", "msg_data": {}}
> 2023-09-11 12:58:07.326893 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":108,"msg":"entity_command","msg_data":{"cmd_id":"cursor_down","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
> 2023-09-11 12:58:06.647171 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 107, "code": 200, "msg": "result", "msg_data": {}}
> 2023-09-11 12:58:06.641784 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":107,"msg":"entity_command","msg_data":{"cmd_id":"cursor_down","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
> 2023-09-11 12:58:06.176987 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 106, "code": 200, "msg": "result", "msg_data": {}}
> 2023-09-11 12:58:06.171262 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":106,"msg":"entity_command","msg_data":{"cmd_id":"cursor_left","entity_id":"D4F8292E8A2C","entity_type":"media_player"}}
> 2023-09-11 12:58:05.508621 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 105, "code": 200, "msg": "result", "msg_data": {}}
> ```
> 
> Rebooting the remote, also helped.

Could you please upload logs for the `Core service` as well when you see this issue? Does it happen after wakeup?

**Comment by Yezariael75** [comment]
Where can I get the core log?

**Comment by splattner** [comment]
> Where can I get the core log?

in the webconfigurator -> settings -> Development -> logs

![Screenshot from 2023-09-11 17-31-52](https://github.com/unfoldedcircle/bug-reports/assets/3875381/669bde07-fbc4-4f34-a9a4-9d97d4d17dea)

**Comment by Yezariael75** [comment]
Here is the "internal server error". After restarting the remote it works again like it should.

2023-09-11 21:41:31.552080 +00:00	core	NOTICE	Setting button backlight to: 0%
2023-09-11 21:41:25.551601 +00:00	core	NOTICE	Changed battery status: 30% Charging, 3936mV, charger: true
2023-09-11 21:41:15.541773 +00:00	core	NOTICE	Changed battery status: 30% Discharging, 3735mV, charger: true
2023-09-11 21:41:11.650729 +00:00	core	ERROR	[uc.main.d4aee80b-638f-42f3-8a88-6fdf3dd64034] error executing activity command uc_androidtv_driver.main.00044BB4FC57:media_player.off (ignore error: false): Internal Server Error
2023-09-11 21:41:11.341920 +00:00	core	NOTICE	[uc.main.d4aee80b-638f-42f3-8a88-6fdf3dd64034] started activity with timeout of 30s
2023-09-11 21:40:55.915513 +00:00	core	ERROR	[uc.main.d4aee80b-638f-42f3-8a88-6fdf3dd64034] error executing activity command uc_androidtv_driver.main.00044BB4FC57:media_player.on (ignore error: false): Internal Server Error
2023-09-11 21:40:55.272552 +00:00	core	NOTICE	[uc.main.d4aee80b-638f-42f3-8a88-6fdf3dd64034] started activity with timeout of 30s
2023-09-11 21:40:50.526892 +00:00	core	NOTICE	Changed battery status: 30% Discharging, 3777mV, charger: false
2023-09-11 21:40:48.727791 +00:00	core	NOTICE	Setting button backlight to: 50%
2023-09-11 21:39:18.554138 +00:00	core	NOTICE	Setting button backlight to: 0%
2023-09-11 21:39:09.959594 +00:00	core	ERROR	Error encountered while processing the incoming HTTP request: AuthError("Authentication error")
2023-09-11 21:39:09.954387 +00:00	core	ERROR	Error encountered while processing the incoming HTTP request: AuthError("Authentication error")
2023-09-11 21:39:02.551790 +00:00	core	NOTICE	Setting button backlight to: 0%
2023-09-11 21:39:00.429749 +00:00	core	NOTICE	Changed battery status: 30% Charging, 3943mV, charger: true
2023-09-11 21:38:50.421353 +00:00	core	NOTICE	Changed battery status: 30% Discharging, 3745mV, charger: true
2023-09-11 21:38:45.550281 +00:00	core	NOTICE	Closing connection with code Policy: Authentication timeout
2023-09-11 21:38:40.413261 +00:00	core	NOTICE	Changed battery status: 30% Discharging, 3754mV, charger: false
2023-09-11 21:38:35.410431 +00:00	core	NOTICE	Changed battery status: 30% Not charging, 3950mV, charger: false
2023-09-11 21:38:34.609558 +00:00	core	NOTICE	Setting button backlight to: 50%
2023-09-11 21:38:16.021306 +00:00	core	NOTICE	No session state has been found for a valid session key, creating a new empty session.
2023-09-11 21:38:15.854202 +00:00	core	ERROR	Error encountered while processing the incoming HTTP request: AuthError("Authentication error")
2023-09-11 21:38:15.850005 +00:00	core	NOTICE	No session state has been found for a valid session key, creating a new empty session.
2023-09-11 21:37:45.367389 +00:00	core	NOTICE	Changed battery status: 30% Charging, 3950mV, charger: true
2023-09-11 21:34:35.204427 +00:00	core	NOTICE	Changed battery status: 29% Charging, 3948mV, charger: true

**Comment by smarthomesydney** [comment]
@martonborzak I have this error also and am using nVidia Shield pro also. 
Let me know if you need additional logs.

**Comment by otherguy** [comment]
I have the same without an Nvidia Shield. The update to the Android TV integration finally let me add my TV, but I‘m facing this issue.

**Comment by smarthomesydney** [comment]
Errors

"2023-09-12 11:35:50.289827 +00:00	core	ERROR	Error("missing field `api`", line: 0, column: 0)"

"2023-09-12 11:35:49.154044 +00:00	core	ERROR	[uc_androidtv_driver] Error connecting to driver: ServiceUnavailable("Failed to connect to host: Connection refused (os error 111)")"

**Comment by Yezariael75** [comment]
@martonborzak  Any news about this? Thanks!

**Comment by kennymc-c** [comment]
Same problem for me but with a Sony XE9005 running Android TV 9

```
2023-10-01 10:04:06.183299 +00:00	core	ERROR	[uc.main.02219a2f-d9f7-4899-91b4-7108d2cf5f6b] error executing activity command uc_androidtv_driver.main.A86BAD280614:media_player.on (ignore error: false): Internal Server Error
2023-10-01 10:04:06.182286 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 32, "code": 500, "msg": "result", "msg_data": {}}
2023-10-01 10:04:06.182059 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: A86BAD280614
2023-10-01 10:04:06.179469 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":32,"msg":"entity_command","msg_data":{"cmd_id":"on","entity_id":"A86BAD280614","entity_type":"media_player","params":{}}}
```

**Comment by vistalba** [comment]
Updated to 1.4.3 and 0.92 and still running in to the same issue.
Just selected to download debug log... but showing only "Info". Looks like this:

```
2023-10-23 18:46:03.106933 +00:00	ui	WARN	uc.ui.entity.controller: Cannot execute command: "media_player.back" 500 "Internal server error, please try later"
2023-10-23 18:46:03.091749 +00:00	ui	WARN	uc.ui.entity.controller: Cannot execute command: "media_player.back" 500 "Internal server error, please try later"
2023-10-23 18:46:03.090768 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 69, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:03.089050 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:03.085708 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":69,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:03.077379 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 68, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:03.076424 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:03.072183 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":68,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:02.606026 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 67, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:02.604488 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:02.601270 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":67,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:02.555870 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 66, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:02.554376 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:02.551174 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":66,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:02.538041 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 65, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:02.536576 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:02.533336 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":65,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:02.305221 +00:00	ui	WARN	uc.ui.entity.controller: Cannot execute command: "media_player.back" 500 "Internal server error, please try later"
2023-10-23 18:46:02.299591 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 64, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:02.298070 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:02.294854 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":64,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:02.087003 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 63, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:02.085566 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:02.082355 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":63,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:02.031405 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 62, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:02.030429 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
2023-10-23 18:46:02.026356 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:<-: {"kind":"req","id":62,"msg":"entity_command","msg_data":{"cmd_id":"back","entity_id":"00044BXYZXYZ","entity_type":"media_player"}}
2023-10-23 18:46:01.909052 +00:00	intg-androidtv	INFO	DEBUG:ucapi.api:->: {"kind": "resp", "req_id": 61, "code": 500, "msg": "result", "msg_data": {}}
2023-10-23 18:46:01.907342 +00:00	intg-androidtv	INFO	ERROR:tv:Cannot send command, connection lost: 00044BXYZXYZ
````

Is there anything I can test or provide helping to nail down this annoying issue?

**Comment by MarkJenniskens** [comment]
I've got a similar issue, maybe I should open a new issue. But every other day the Android integration stops working (time-out / internal server error). Sometimes rebooting the remote helps. When it doesn't i need te re-run the integration set-up to pair with the Nvidia Shield. I can reproduce the issue by powercycling the Shield.

**Comment by zehnm** [comment]
We found one reconnection related issue, which is fixed in the next release. We continue testing though to fix this showstopper for good.

**Comment by vistalba** [comment]
> We found one reconnection related issue, which is fixed in the next release. We continue testing though to fix this showstopper for good.

Thanks for the update. Any ETA?
One more thing I noticed… for me the Bug just occur after the Remote was in Sleep and most probably disconnected from WiFi. So may it is related to WiFi/Remote combination? My remote does most of the time not work after waking up for 3-4 seconds. 
Using UniFi AP with multiple APs. Is it normal, that the remote needs that long to reconnect?

**Comment by zehnm** [comment]
Closing: fixed in release 1.4.5.
We were no longer able to reproduce this issue with Nvidia Shield TV.
If you still encounter reconnect issues after wakeup, please reopen this issue.

**Comment by vistalba** [comment]
> Closing: fixed in release 1.4.5. We were no longer able to reproduce this issue with Nvidia Shield TV. If you still encounter reconnect issues after wakeup, please reopen this issue.

After 3 days I can say that it is defintly better than before. I still run into some issues like:

- after wakeup it takes 2-4 seconds until I can use the remote
- after pressing a button, nothing happens and after a few seconds a message appears "internal server error" in red. Most of the time the issue does now recover from that in a few seconds. But it is still annyoing for me.
